Sources/Commands/SwiftPackageTool.swift

```diff
@@ -896,6 +899,226 @@ extension SwiftPackageTool {
         }
     }
 
+    struct InstallPackage: SwiftCommand {
+        static let configuration: CommandConfiguration = CommandConfiguration(
+            abstract: "Clone, compile, and copy a remote executable target"
+        )
+        
+        enum RepoAccessType: String, EnumerableFlag {
+            case url, path
+        }
+        
+        @Flag(help: "The way to access the given repo, either by a remote git URL or a on-disk path")
+        var accessType: RepoAccessType
+        
+        @Argument(help: "URL of the remote target if using --url, or path to the repo when using --path")
+        var repoPathOrURL: String
```

Comment:
IMO for an initial implementation it's better to always operate on `Package.swift` that's already in cwd or specified by `--package-path` like the rest of `swift package` subcommands do. I suggest removing these arguments and flags and just assuming this subcommand always runs where a package manifest to parse is available.

Reply:
Noted for rewrite

Reply:
@MaxDesiatov @SerenaKit would be interesting to consider a `swift package install <scm url>` and `swift package install <registry identifier>`, otherwise it would be a two step process

Reply:
I totally support both of those options for installing remote packages, they are much more complex and will take more time to get right though. To keep iterations small it would be great to start with a relatively small implementation for installing locally cloned repositories, get that reviewed and merged under an `experimental-` prefix if we agree on this direction in principle.

Then we can implement installation of remote repositories in a subsequent iteration, depending on community feedback in the pitch thread.

Reply:
Also we would first need to redesign SwiftPM's command structure, `swift package install` cannot install a remote package today by definition (since `swift package` implies being in a package).